buildSrc/src/main/groovy/io/deephaven/tools/docker/DeephavenInDockerExtension.groovy

```diff
@@ -82,6 +97,14 @@ public abstract class DeephavenInDockerExtension {
             task.containerId.set containerName.get()
         }
 
+        portTask = project.tasks.register('waitForPort', DockerInspectContainer) {task ->
+            task.dependsOn healthyTask
+            task.containerId.set containerName.get()
+            task.onNext { InspectContainerResponse inspect ->
+                getPort().set(Integer.parseInt(((InspectContainerResponse) inspect).getNetworkSettings().ports.bindings.values().first()[0].hostPortSpec))
+            }
+        }
```

Comment:
Is this meant to be an ad-hoc / user-run task? I don't see anything that depends on it.

I'm assuming the end-2-end idea is that we can instead do:

```diff
diff --git a/py/client/build.gradle b/py/client/build.gradle
index 2f61f8b30c..68511d713a 100644
--- a/py/client/build.gradle
+++ b/py/client/build.gradle
@@ -93,7 +93,7 @@ tasks.getByName('check').dependsOn(Docker.registerDockerTask(project, 'testPyCli
             into 'project/'
         }
     }
-    containerDependencies.dependsOn = [deephavenDocker.healthyTask]
+    containerDependencies.dependsOn = [deephavenDocker.portTask]
     containerDependencies.finalizedBy = deephavenDocker.endTask
     network = deephavenDocker.networkName.get()
     dockerfile {
@@ -105,7 +105,7 @@ tasks.getByName('check').dependsOn(Docker.registerDockerTask(project, 'testPyCli
                       pip3 install --upgrade pip; \\
                       pip3 install -r requirements-dev.txt'''
         environmentVariable 'DH_HOST', deephavenDocker.containerName.get()
-        environmentVariable 'DH_PORT', '10000'
+        environmentVariable 'DH_PORT', "${deephavenDocker.getPort()}"
     }
     parentContainers = [ Docker.registryTask(project, 'python') ]
     entrypoint = ['python', '-m', 'xmlrunner', 'discover', 'tests', '-v', '-o', '/out/report']
```

Reply:
That's the idea, but what you have there won't work since the python client tests are run in _another_ docker container, and the getPort() exposed here is the exposed port on the host, not the container.

Instead, this can be used for JS API tests, or java-client tests, in pretty much the way that you described above.